### PR TITLE
fix: error handing, add e2e tests

### PR DESCRIPTION
**Pull request by toddbaert**
* fixes error handling of in-process provider
  * the json-eval returns simple errors (`errors.new(..)`) not ResolutionErrors
* improves the flexibility of the flagd test suite (allows you to supply a provider factory instead of creating a particular provider
* implements the basic `evaluation.feature` test in the in-process provider (which works great!)

### CODE REVIEW

File: providers/flagd-in-process/pkg/provider.go [hunk 1]
```diff
@@ -646,3 +624,16 @@ func syncProviderFromConfig(logger *logger.Logger, sourceConfig runtime.SourceCo
 			sourceConfig.Provider, SourceTypeGrpc, SourceTypeKubernetes)
 	}
 }
+
+func mapError(err error) of.ResolutionError {
```
Comment (toddbaert) [inline providers/flagd-in-process/pkg/provider.go:628]: @bacherfl @Kavindu-Dodan this is the only functional change in this PR - as the description says, the errors thrown in the json-evaluator are just errors, so they need to be handled this way (flagd itself has similar code for mapping them to connect errors).

File: tests/flagd/pkg/integration/evaluation.go [hunk 2]
```diff
@@ -7,12 +7,11 @@ import (
 	"strconv"
 
 	"github.com/cucumber/godog"
-	flagd "github.com/open-feature/go-sdk-contrib/providers/flagd/pkg"
 	"github.com/open-feature/go-sdk/pkg/openfeature"
 )
 
-func InitializeEvaluationScenario(pOptions ...flagd.ProviderOption) func(*godog.ScenarioContext) {
-	providerOptions = pOptions
+func InitializeEvaluationScenario(providerSupplier func() openfeature.FeatureProvider) func(*godog.ScenarioContext) {
```
Comment (toddbaert) [inline tests/flagd/pkg/integration/evaluation.go:13]: The test suites now take a provider factory (so you can use any provider with them).

File: providers/flagd-in-process/pkg/provider.go (hunk 1 above)
Comment (Kavindu-Dodan) [reply providers/flagd-in-process/pkg/provider.go:628]: Yup, this fully matches with flagd error handling - https://github.com/open-feature/flagd/blob/main/core/pkg/service/flag-evaluation/flag_evaluator.go#L316

**Review by Kavindu-Dodan — APPROVED** [review]
changes lgtm

**Review by bacherfl — APPROVED** [review]

**Pull request merged**